redis_json/src/commands.rs

```diff
@@ -624,16 +618,12 @@ pub fn json_mget<M: Manager>(manager: M, ctx: &Context, args: Vec<RedisString>)
                         json_key.get_value().map_or(Ok(RedisValue::Null), |value| {
                             value.map_or(Ok(RedisValue::Null), |doc| {
                                 let key_value = KeyValue::new(doc);
-                                if format_options.is_resp3_reply() {
-                                    Ok(key_value.to_resp3_path(&path, &format_options))
+                                let res = if !path.is_legacy() {
```

Comment:
`to_resp3_path` is not needed since with `ReplyFormat::STRING` it is same as the to_string_  methods.